python/ray/util/iter.py

```diff
@@ -1059,6 +1112,7 @@ def build_union(timeout=None):
                 yield_counts = [0] * len(active)
                 removed_iter_indices = []
                 for i, (weight, it) in enumerate(list(active)):
+                    expected_yield_counts = weight if weight != "*" else MAX_PULL
```

Comment:
I don't quite follow why we move it from the case on `StopIteration`. I can sync up with you offline.